edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/Avaliability.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Metrics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class Avaliability
+    {
+        private readonly ISystemTime time;
+
+        public string Name;
+        public string Version;
+        public TimeSpan TotalTime = TimeSpan.Zero;
+        public TimeSpan Uptime = TimeSpan.Zero;
+
+        private DateTime? previousMeasure = null;
+
+        public Avaliability(string name, string version, ISystemTime time)
+        {
+            this.Name = name;
+            this.Version = version;
+
+            this.time = time;
+            this.previousMeasure = time.UtcNow;
+        }
+
+        public Avaliability(AvaliabilityRaw raw, ISystemTime time)
+        {
+            this.Name = raw.Name;
+            this.Version = raw.Version;
+            this.Uptime = raw.Uptime;
+            this.TotalTime = raw.TotalTime;
+
+            this.time = time;
+        }
```

Comment:
why this.previousMeasure is not initialized in this constructor?

Reply:
This constructor is used when the availability is loaded from a file. It doesn't take a measurement when it is loaded from a file, so previous measure is still null.

Reply:
should it be frozen/readonly if it is loaded from a file?

Reply:
no, it saves itself to a file on shutdown, and re-loads itself on startup. This is so all modules availabilities won't reset when agent restarts.